src/main/java/duke/command/DateCommand.java

```diff
@@ -65,6 +65,43 @@ public void execute(TaskList taskList, Storage storage) {
         }
     }
 
+    /**
+     * Executes Date behaviour of finding all tasks on a given date
+     * and returns the result as a String
+     * @param taskList to access collection of tasks.
+     * @param storage to access save-load functionality.
+     * @return a String representing the output.
+     */
+    public String executeWithBotResponse(TaskList taskList, Storage storage) {
+        String output = "";
+
+        try {
+            LocalDate date =  LocalDate.parse(dateStr);
+            int count = 0;
+            for (Task t: taskList.getList()) {
+                if (t instanceof Deadline) {
```

Comment:
Is using instanceOf best practice for OOP? Consider other methods of doing this.